tests/test_sequence.py

```diff
@@ -1074,16 +1074,33 @@ def test_switch_device_up(
     assert "digital" in seq.switch_device(devices[1], True).declared_channels
 
 
+@pytest.mark.parametrize("device_type", ["analog", "extended"])
 @pytest.mark.parametrize("mappable_reg", [False, True])
 @pytest.mark.parametrize("parametrized", [False, True])
-@pytest.mark.parametrize("extension_arg", ["amp", "control", "buffer_time"])
+@pytest.mark.parametrize(
+    "extension_arg", ["amp", "control", "2control", "buffer_time"]
+)
 def test_switch_device_eom(
-    reg, mappable_reg, parametrized, extension_arg, patch_plt_show
+    reg, device_type, mappable_reg, parametrized, extension_arg, patch_plt_show
 ):
+    if device_type == "analog":
+        device = AnalogDevice
+    elif device_type == "extended":
+        eom = dataclasses.replace(
+            AnalogDevice.channels["rydberg_global"].eom_config,
+            controlled_beams=tuple(RydbergBeam),
+            multiple_beam_control=True,
+            custom_buffer_time=None,
+        )
+        channel = dataclasses.replace(
+            AnalogDevice.channels["rydberg_global"], eom_config=eom
+        )
+        device = dataclasses.replace(AnalogDevice, channel_objects=(channel,))
```

Comment:
nit: you could do this outside the function and just give the `device` as an argument, no?

Reply:
Sure :)